x-pack/plugins/apm/public/selectors/latency_chart_selectors.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import { rgba } from 'polished';
+import { EuiTheme } from '../../../observability/public';
+import { asDuration } from '../../common/utils/formatters';
+import {
+  Coordinate,
+  RectCoordinate,
+  TimeSeries,
+} from '../../typings/timeseries';
+import { APIReturnType } from '../services/rest/createCallApmApi';
+
+export type LatencyChartsResponse = APIReturnType<'GET /api/apm/services/{serviceName}/transactions/charts/latency'>;
+
+interface LatencyChart {
+  latencyTimeseries: TimeSeries[];
+  mlJobId?: string;
+  anomalyTimeseries?: {
+    bounderies: TimeSeries;
+    scores: TimeSeries;
+  };
+}
+
+export function getLatencyChartSelector({
+  latencyChart,
+  theme,
+}: {
+  latencyChart?: LatencyChartsResponse;
+  theme: EuiTheme;
+}): LatencyChart {
+  if (!latencyChart) {
+    return {
+      latencyTimeseries: [],
+      mlJobId: undefined,
+      anomalyTimeseries: undefined,
```

Comment:
I think we might want to break latency and ML into separate APIs as well. We can leave that for now though.

Reply:
That's my plan too but thought that this PR is already big enough.